core/src/translate/query.rs

```diff
@@ -131,18 +145,84 @@ fn translate_table_with_joins(sql_table_with_joins: &SqlTableWithJoins) -> Resul
+                SqlValue::Number(big_decimal, _),
+            )))) => size_from(big_decimal),
+            Some(FunctionArg::Unnamed(FunctionArgExpr::Expr(SqlExpr::UnaryOp { op, expr }))) => {
+                match (op, expr.as_ref()) {
+                    (UnaryOperator::Minus, SqlExpr::Value(SqlValue::Number(big_decimal, _))) => {
+                        let size = size_from(big_decimal)?;
+
+                        Err(TranslateError::WrongSeriesSize(-size).into())
+                    }
+                    (UnaryOperator::Plus, SqlExpr::Value(SqlValue::Number(big_decimal, _))) => {
+                        size_from(big_decimal)
+                    }
+                    _ => Err(TranslateError::UnsupportedArgsUnaryOp {
+                        op: format!("{op}"),
+                        expr: format!("{expr}"),
+                    }
+                    .into()),
+                }
+            }
+            None => Err(TranslateError::LackOfSeriesSize.into()),
+            _ => Err(TranslateError::UnsupportedArgs(format!(
+                "{}",
+                function_args
+                    .iter()
+                    .map(|v| v.to_string())
+                    .collect::<Vec<_>>()
+                    .join(", ")
+            ))
+            .into()),
+        },
+        None => Err(TranslateError::LackOfArgs.into()),
+    }
+}
+
 fn translate_table_factor(sql_table_factor: &SqlTableFactor) -> Result<TableFactor> {
     match sql_table_factor {
-        SqlTableFactor::Table { name, alias, .. } => Ok(TableFactor::Table {
-            name: translate_object_name(name),
-            alias: alias
-                .as_ref()
-                .map(|SqlTableAlias { name, columns }| TableAlias {
-                    name: name.value.to_owned(),
-                    columns: translate_idents(columns),
-                }),
-            index: None, // query execution plan
-        }),
+        SqlTableFactor::Table {
+            name, alias, args, ..
+        } if get_name(&translate_object_name(name))?.to_uppercase() == "SERIES"
+            && args.is_some() =>
+        {
+            Ok(TableFactor::Series {
+                name: translate_object_name(name),
+                alias: alias
+                    .as_ref()
+                    .map(|SqlTableAlias { name, columns }| TableAlias {
+                        name: name.value.to_owned(),
+                        columns: translate_idents(columns),
+                    }),
+                size: translate_table_args(args)?,
+            })
+        }
+        SqlTableFactor::Table { name, alias, .. } => {
+            Ok(TableFactor::Table {
+                name: translate_object_name(name),
+                alias: alias
+                    .as_ref()
+                    .map(|SqlTableAlias { name, columns }| TableAlias {
+                        name: name.value.to_owned(),
+                        columns: translate_idents(columns),
+                    }),
```

Comment:
There exists two translate code of `SqlTableAlias` to `TableAlias`.
It looks good to add `translate_table_alias` function to handle this.